src/mmu.rs

```diff
@@ -903,6 +899,26 @@ impl PageTable {
             self.pml4.map(P::mapping(page, frame, attrs));
         }
     }
+
+    /// XXX(cross): Make this actually walk the table to make sure
+    /// the VA is really mapped.
+    fn is_mapped(&self, va: usize) -> bool {
+        va != 0
+    }
+
+    /// Returns a raw pointer to a virtual address mapped by
+    /// this table.
+    pub(crate) fn try_with_addr<T>(&self, va: usize) -> Result<*mut T, Error> {
+        if !self.is_mapped(va) {
+            return Err(Error::BadPointer);
+        }
+        let base = self as *const _ as *const u8;
+        let ptr = base.with_addr(va);
```

Comment:
I think this should use from_exposed_addr instead. The provenance of base covers the page table itself, not whatever it maps.

Reply:
Done.